src/main/scala/SearchTweets.scala

```diff
@@ -16,24 +16,49 @@ import java.io.IOException;
 import java.net.URISyntaxException;
 import scala.collection.mutable.ArrayBuffer
 
+import scala.util.parsing.json._
 
 object SearchTweets {
 
   @throws[IOException]
   def main(args: Array[String]): Unit = {
 
-    var bearerToken: String = "Sundoo"
+    var bearerToken: String = "YOUR API KEY HERE"
 
     if (bearerToken != null) {
           Future {
 
-      var response: String = getTweets("BurgerKing", bearerToken)
-      println(response)
+            var response: String = getTweets("BurgerKing", bearerToken)
+            println(response)
+            //deconstruct JSON object to find next_token
+            val json:Option[Any] = JSON.parseFull(response)
+            val map:Map[String,Any] = json.get.asInstanceOf[Map[String, Any]]
+            val meta:Map[String, String] = map.get("meta").get.asInstanceOf[Map[String, String]]
+            //if there is next_token, call retrievePages function to recursively retrieve all pages
+            if (meta.contains("next_token")) {
+              val next_token = meta("next_token")
+              retrievePages(next_token)
+            }
           }
     } else {
       println("There was a problem getting you bearer token. Please make sure you set the BEARER_TOKEN environment variable")
     }
-    Thread.sleep(3000)
+    Thread.sleep(10000)
+  }
+
+  //recursively retrieve all pages from API endpoint, printing them to console
+  def retrievePages(next_token: String): Unit = {
+      var bearerToken: String = "AAAAAAAAAAAAAAAAAAAAAKUnLwEAAAAApWuEw9K8SVH7%2BqNNxVtg36Ryr5o%3DP3sGYZngNVtApjFYqjMC5r3lTg8cF4gGIdCtcjUJAjnd23VpEk"
```

Comment:
A variable bearerToken is already used at line 26, you should pass that variable as a parameter in this function

Reply:
Ah yeah I meant to do this but forgot thanks